### PR TITLE
Material and Mesh imGUI importer

### PR DESCRIPTION
**Pull request by AuxPort66**
- Add Mesh imGUI, remove and import file browser
- Add Material imGUI functionality
- Add material delete and import file browser
- It was necessary to add some basic functionality in the file system flow (Update .bin when Resource changes and special case for Material because we also changed the .mat of Assets/Materials)

### CODE REVIEW

File: Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		switch (textureType)
+		{
+		case TextureType::DIFFUSE:
+			materialAsShared->SetDiffuseUID(uidTexture);
+			break;
+		case TextureType::NORMAL:
+			materialAsShared->SetNormalUID(uidTexture);
+			break;
+		case TextureType::OCCLUSION:
+			break;
```
Comment (markmeliaa) [inline Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp:27]: Why the occlusion is not set? Is this alright?

File: Source/DataModels/Components/ComponentMaterial.cpp [hunk 2]
```diff
@@ -40,80 +41,280 @@ void ComponentMaterial::Draw()
...
+				}
+			}
+		}
+		//bool hasNormal;
+		//std::shared_ptr<ResourceTexture> textureAsShared = textureNormal.lock();
+		//textureAsShared ? hasNormal = true : hasNormal = false;
+
+		//ImGui::Checkbox("Normal slider", &hasNormal);
+		ImGui::SliderFloat("Normal", &normalStrength, 0.0f, 1.0f, "%.1f", ImGuiSliderFlags_AlwaysClamp);
+		ImGui::Separator();
```
Comment (markmeliaa) [inline Source/DataModels/Components/ComponentMaterial.cpp:313]: this ImGUI::Separator should not be here

File: Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp (hunk 1 above)
Comment (AuxPort66) [reply Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp:27]: We haven't processed the occlusion yet, that's why it's empty

File: Source/DataModels/Components/ComponentMaterial.cpp [hunk 3]
```diff
@@ -40,80 +41,280 @@ void ComponentMaterial::Draw()
...
+
+		ImGui::Separator();
```
Comment (markmeliaa) [inline Source/DataModels/Components/ComponentMaterial.cpp:315]: And this ImGui separator should be outside the }

File: Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp (hunk 1 above)
Comment (markmeliaa) [reply Source/DataModels/Windows/EditorWindows/ImporterWindows/WindowTextureInput.cpp:27]: oke!

**Comment by markmeliaa** [comment]
Everything is good, just a couple separators for aesthetic purposes and that's all.

**Pull request merged**